app/forms/attached_files_populator.rb

```diff
@@ -9,15 +9,15 @@ def call(form, fragment:, **)
     item = existing_record(form: form, id: fragment['id'])
 
     if fragment['_destroy'] == '1'
-      # Remove AttachedFile and associated AS model instances if AF exists
-      # Else, there is no AF, so remove the AS::Blob directly
-      if item
-        form.attached_files.delete(item)
-      else
-        ActiveStorage::Blob.find_signed!(fragment['file']).purge_later
```

Comment:
Nice to see this go.

Reply:
Yes, we don't want a model serializer to be responsible for doing persistence.